client/iOS/FreeRDP/ios_cliprdr.m

```diff
@@ -0,0 +1,503 @@
+/**
+ * FreeRDP: A Remote Desktop Protocol Implementation
+ * Android Clipboard Redirection
+ *
+ * Copyright 2013 Felix Long
+ * Copyright 2015 Thincast Technologies GmbH
+ * Copyright 2015 DI (FH) Martin Haimberger <martin.haimberger@thincast.com>
+ * Copyright 2023 Iordan Iordanov
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <freerdp/config.h>
+
+#include <winpr/crt.h>
+#include <winpr/stream.h>
+
+#include <freerdp/client/channels.h>
+#include <freerdp/client/cliprdr.h>
+
+#include "ios_cliprdr.h"
+
+UINT ios_cliprdr_send_client_format_list(CliprdrClientContext* cliprdr)
+{
+	UINT rc = ERROR_INTERNAL_ERROR;
+	UINT32 index;
+	UINT32 formatId;
+	UINT32 numFormats;
+	UINT32* pFormatIds;
+	const char* formatName;
+	CLIPRDR_FORMAT* formats;
+	CLIPRDR_FORMAT_LIST formatList = { 0 };
+
+	if (!cliprdr)
```

Comment:
It would not be an error due to invalid input, but wrong API usage. Use an assertion (`WINPR_ASSERT()`) instead.

Reply:
Hey Pascal, i am not sure I am familiar enough with FreeRDP to make this kind of change and actually ensure that it works. The majority of this implementation was simply taken from android_cliprdr.c. I think your comments are valuable but maybe we should take them up in a separate PR that refactors both ios_cliprdr.c and android_cliprdr.c. What do you think?

Reply:
@iiordanov not a blocking thing, but yes, the clients do some unnecessary checks. we moved to using assertions for required arguments some time ago, but did not have the time to go through the whole code base.
so, if you´d do that that would be great (and to do it in another pr most likely the best way to go)